### PR TITLE
Fix all pylint errors and add scripts that make it easy to lint and test locally

### PR DESCRIPTION
**Pull request by matthewkol186**
Big refactor of our codebase that doesn't really change functionality but hopefully makes everything a bit cleaner and easier to understand. 
- Remove unnecessary imports
- Remove duplicated code
- Refactor input validation in recommender models to a common function
- Require all user inputs to be consistent (i.e., don't ignore `num_users` if `user_representation` is passed in as well, instead enforce that `num_users` and `user_representation` must be consistent with one another)
- Add new tests for new input validation system

### CODE REVIEW

**Review by elucherini — CHANGES_REQUESTED** [review]
This looks great, thanks! I left a few comments in the code -- mainly questions about some decisions.

One main thing that I'd like to see is: `rec` must become `trecs`. Similarly, every reference to `algo-segregation` must change to `t-recs`. Feel free to address this in another PR if you think it's more reasonable, but I figured this would be a good place since you've already done a lot of spring cleaning.

File: rec/components/users.py [hunk 1]
```diff
@@ -173,17 +172,14 @@ def __init__(
         # this will be initialized by the system
         self.actual_user_scores = actual_user_scores
         if num_users is not None:
-            self._user_vector = np.arange(num_users, dtype=int)
+            self.user_vector = np.arange(num_users, dtype=int)
```
[the remark this answers is not in the document]
Comment (matthewkol186) [reply rec/components/users.py:175]: b/c it wasn't actually an internal variable 😢  see it gets used by many external classes
image

File: rec/metrics/measurement.py [hunk 2]
```diff
@@ -99,15 +154,15 @@ class InteractionMeasurement(Measurement):
...
 
-    def __init__(self, verbose=False):
-        self.name = "interaction_histogram"
-        Measurement.__init__(self, verbose, init_value=None)
+    def __init__(self, name="interaction_histogram", verbose=False):
+        Measurement.__init__(self, name, verbose, init_value=None)
 
-    def _generate_interaction_histogram(self, interactions, num_users, num_items):
+    @staticmethod
+    def generate_interaction_histogram(interactions, num_users, num_items):
```
[the remark this answers is not in the document]
Comment (matthewkol186) [reply rec/metrics/measurement.py:165]: not sure why it was changed! can change it back

File: rec/components/users.py (hunk 1 above)
[the remark this answers is not in the document]
Comment (elucherini) [reply rec/components/users.py:175]: lmao true ok sorry

**Review by elucherini — APPROVED** [review]
Thanks!!

**Pull request merged**